src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1045,6 +1045,22 @@ internal void DeleteModelInternal(List<ModelBase> modelsToDelete)
             if (null == CurrentWorkspace)
                 return;
 
+            //Check for empty group
+            var annotations = Workspaces.OfType<HomeWorkspaceModel>().SelectMany(ws => ws.Annotations);
+            foreach (var annotation in annotations)
+            {
+                if (!annotation.SelectedModels.Except(modelsToDelete).ToList().Any())
+                {
+                    //Annotation Model has to be serialized first - before the nodes.
+                    //so, store the Annotation model as first object. This will serialize the 
+                    //annotation before the nodes are deleted. So, when Undo is pressed,
+                    //annotation model is deserialized correctly.
+                    var models = new LinkedList<ModelBase>(modelsToDelete);
+                    models.AddFirst(annotation);
+                    modelsToDelete = models.ToList();
```

Comment:
Yes. Because I want the annotation model to be removed first before the nodes/notes are removed. otherwise, the undo operation behaves different. In LinkedList, inserting to the first is not very expensive operation.